overlord/snapstate/snapstate.go

```diff
@@ -1243,6 +1248,7 @@ func Switch(st *state.State, name string, opts *RevisionOptions) (*state.TaskSet
 	snapsup := &SnapSetup{
 		SideInfo:    snapst.CurrentSideInfo(),
 		Channel:     opts.Channel,
+		CohortKey:   opts.CohortKey,
```

Comment:
or this should be snapst.CohortKey if opts.CohortKey is ""

Reply:
yes.